aten/src/ATen/core/ivalue_inl.h

```diff
@@ -23,6 +23,10 @@ namespace c10 {
 struct IValue;
 struct ClassType;
 struct TupleType;
+template <typename T>
```

Comment:
When creating the function closure for `__init__`, I was passing in a function that looks like this:
```[](IValue x, args...)```

(We need to pass in an IValue as I couldn't find a way to construct an `ivalue::object`, and modify the `capsule` attribute as well).

However, `kernel_functor.h` automatically sets the input types of the function by introspecting the types of the functor (which is `IValue` in this case). However, we need the type it binds to be `Foo`. 

Thus, instead, I pass in a lambda
```[](ivalue_holder<CurClass> x, args...)```
from which `getTypePtr` can recover the original class pointer.

Reply:
Ok, can you add this response as a comment. Also `tagged_capsule` might be a better name since it is more specific to its use. Replacing T with `CapsuleClassType` might make things clearer as well.